ttnn/cpp/ttnn/operations/moreh/moreh_norm/device/moreh_norm_program_factory_h.cpp

```diff
@@ -0,0 +1,230 @@
+        "reader_moreh_norm_h.cpp";
+    const auto writer_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_norm/device/moreh_norm_h/kernels/"
+        "writer_moreh_norm_h.cpp";
+
+    const auto reader_kernels_id = tt::operations::primary::CreateReadKernel(program, reader_kernel_file, all_cores);
+    const auto writer_kernels_id = tt::operations::primary::CreateWriteKernel(program, writer_kernel_file, all_cores);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      ComputeKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    std::map<std::string, std::string> compute_defines{};
+    compute_defines["REDUCE_OP"] = "PoolType::SUM";
+    compute_defines["REDUCE_DIM"] = "ReduceDim::REDUCE_COL";
+
+    const auto compute_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_norm/device/moreh_norm_h/kernels/"
+        "moreh_norm_h_kernel.cpp";
+
+    const auto compute_kernels_id_1 = tt::operations::primary::CreateComputeKernel(
+        program,
+        compute_kernel_file,
+        {core_group_1, num_units_per_core_group_1},
+        compute_defines,
+        math_fidelity,
+        fp32_dest_acc_en,
+        math_approx_mode);
+
+    KernelHandle compute_kernels_id_2{0};
+    if (!core_group_2.ranges().empty()) {
+        compute_kernels_id_2 = tt::operations::primary::CreateComputeKernel(
+            program,
+            compute_kernel_file,
+            {core_group_2, num_units_per_core_group_2},
+            compute_defines,
+            math_fidelity,
+            fp32_dest_acc_en,
+            math_approx_mode);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      RuntimeArgs SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    for (uint32_t i = 0, tile_offset = 0; i < num_cores_to_be_used; ++i) {
+        CoreCoord core = {i / num_cores_y, i % num_cores_y};
+
+        uint32_t num_cols_per_core;
+        KernelHandle compute_kernel_id;
+        if (core_group_1.core_coord_in_core_ranges(core)) {
+            num_cols_per_core = num_units_per_core_group_1;
+            compute_kernel_id = compute_kernels_id_1;
+        } else if (core_group_2.core_coord_in_core_ranges(core)) {
+            num_cols_per_core = num_units_per_core_group_2;
+            compute_kernel_id = compute_kernels_id_2;
+        } else {
+            TT_THROW("Core not in specified core ranges.");
+        }
+
+        // reader
+        const std::vector<uint32_t> reader_runtime_args{
+            input.buffer()->address(),
+            static_cast<uint32_t>(tt::operations::primary::is_dram(input)),
+            *reinterpret_cast<uint32_t*>(&decimal),
+            *reinterpret_cast<uint32_t*>(&recip_p_decimal),
+            num_cols_per_core,
+            tile_offset,
+            Ht,
+            Wt,
+            origin_h};
```

Comment:
I think that you need to override these variables in ```override_runtime_arguments``` too.
Isn't it?

Reply:
Maybe, I could be wrong.

Reply:
The deprecated version didn't override them in override_runtime_arguments_callback, so I didn't do it either, as this is just a porting operation.
The test cases also did not run into that scenario, so I think the acceptance for this PR should be based on passing all test cases, the same as the deprecated version.
New test cases for overriding runtime arguments will be added in another PR.

Reply:
> The deprecated version didn't override them in override_runtime_arguments_callback, so I didn't do it either, as this is just a porting operation. The test cases also did not run into that scenario, so I think the acceptance for this PR should be based on passing all test cases, the same as the deprecated version. New test cases for overriding runtime arguments will be added in another PR.

Thank you for your detailed explanation.